manager/controllers/utils.go

```diff
@@ -257,16 +257,22 @@ func ApplyTagsFilter(c *gin.Context, tx *gorm.DB, systemIDExpr string) (*gorm.DB
 	tags := c.QueryArray("tags")
 	for _, t := range tags {
 		matches := tagRegex.FindStringSubmatch(t)
-		if len(matches) == 4 {
-			tagJSON := fmt.Sprintf(`[{"namespace":"%s", "key": "%s", "value": "%s"}]`, matches[1], matches[2], matches[3])
-			subq = subq.Where(" h.tags @> ?::jsonb", tagJSON)
-			applied = true
-		} else {
+		if len(matches) < 5 {
 			// invalid tag
 			err := errors.Errorf(InvalidTagMsg, t)
 			LogAndRespBadRequest(c, err, err.Error())
 			return tx, false, err
 		}
+		if matches[4] != "" {
+			tagJSON := fmt.Sprintf(`[{"namespace":"%s", "key": "%s", "value": "%s"}]`, matches[1], matches[2], matches[4])
+			subq = subq.Where(" h.tags @> ?::jsonb", tagJSON)
+			applied = true
```

Comment:
These two lines are identical in both "if" branches so we can exclude it from if block and call it after that, right?

Reply:
Fixed